central/plugins/org.jboss.tools.central/src/org/jboss/tools/central/editors/browser/CentralBrowserErrorWrapper.java

```diff
@@ -0,0 +1,100 @@
+package org.jboss.tools.central.editors.browser;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.SWTError;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.swt.events.MouseListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.browser.IWorkbenchBrowserSupport;
+import org.eclipse.ui.internal.part.StatusPart;
+import org.jboss.tools.central.JBossCentralActivator;
+import org.jboss.tools.central.Messages;
+
+public class CentralBrowserErrorWrapper {
+
+	/**
+	 * Logs given {@code throwable} (may be wrapped) and shows error message in 
+	 * the {@code parent} composite.
+	 */
+	public void showError(Composite parent,
+			Throwable originalThrowable) {
+		Throwable throwable = wrapError(originalThrowable);
+		String errorMessage = throwable.getMessage();
+		
+		parent.setLayout(new GridLayout());
+		Composite statusComposite = new Composite(parent, SWT.NONE);
+		Color bgColor= parent.getDisplay().getSystemColor(SWT.COLOR_LIST_BACKGROUND);
+		Color fgColor= parent.getDisplay().getSystemColor(SWT.COLOR_LIST_FOREGROUND);
+		parent.setBackground(bgColor);
+		parent.setForeground(fgColor);
+		GridData gridData = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gridData.widthHint = 0;
+		gridData.heightHint = 0;
+		statusComposite.setLayoutData(gridData);
+
+		IStatus displayStatus = new Status(IStatus.ERROR, JBossCentralActivator.PLUGIN_ID, errorMessage, throwable);
+		new StatusPart(statusComposite, displayStatus);
+		
+		final Link link = new Link(parent, SWT.WRAP);
+		link.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+		link.setBackground(bgColor);
+		link.setText(Messages.visualEditorFaq);
+		link.setToolTipText(Messages.visualEditorFaqLink);
+		link.setForeground(link.getDisplay().getSystemColor(SWT.COLOR_BLUE));
+
+		link.addMouseListener(new MouseListener() {
+			public void mouseDown(org.eclipse.swt.events.MouseEvent e) {
+				BusyIndicator.showWhile(link.getDisplay(), new Runnable() {
```

Comment:
See if you could use JBossCentralActivator.openUrl(location, shell, external) instead.